src/lib/__tests__/date.test.ts

```diff
@@ -1,51 +1,271 @@
+      expect(period).toBe("Dec 5 at 8:00pm")
+    })
+
+    it("includes single date in different year from now", () => {
+      const period = singleDate("2021-12-05T20:00:00+00:00")
+      expect(period).toBe("Dec 5, 2021 at 8:00pm")
+    })
+  })
+
+  describe("singleDateWithoutHour", () => {
+    const realNow = Date.now
+    beforeEach(() => {
+      Date.now = () => new Date("2018-01-30T03:24:00") as any
+    })
+    afterEach(() => {
+      Date.now = realNow
+    })
+
+    it("includes single date in this year", () => {
+      const period = singleDateWithoutHour("2018-12-05T20:00:00+00:00")
+      expect(period).toBe("Wed, Dec 5")
+    })
+
+    it("includes single date in different year from now", () => {
+      const period = singleDateWithoutHour("2021-12-05T20:00:00+00:00")
+      expect(period).toBe("Sun, Dec 5, 2021")
+    })
+  })
+
+  describe("datesWithHours", () => {
+    const realNow = Date.now
+    beforeEach(() => {
+      Date.now = () => new Date("2018-01-30T03:24:00") as any
+    })
+    afterEach(() => {
+      Date.now = realNow
+    })
+
+    it("includes the start and end date with different years", () => {
+      const period = datesWithHours(
+        "2021-12-05T20:00:00+00:00",
+        "2022-12-30T17:00:00+00:00"
+      )
+      expect(period).toBe("Dec 5, 2021 at 8:00pm – Dec 30, 2022 at 5:00pm")
+    })
+
+    it("includes the start and end date with different year from now", () => {
+      const period = datesWithHours(
+        "2020-12-05T20:00:00+00:00",
+        "2020-12-30T17:00:00+00:00"
+      )
+      expect(period).toBe("Dec 5, 2020 at 8:00pm – Dec 30, 2020 at 5:00pm")
+    })
+
+    it("includes the start and end date in current year", () => {
+      const period = datesWithHours(
+        "2018-12-05T20:00:00+00:00",
+        "2018-12-30T17:00:00+00:00"
+      )
+      expect(period).toBe("Dec 5 at 8:00pm – Dec 30 at 5:00pm")
+    })
+
+    it("date is same day in different year from now", () => {
+      const period = datesWithHours(
+        "2019-12-05T20:00:00+00:00",
+        "2019-12-05T22:00:00+00:00"
+      )
+      expect(period).toBe("Dec 5, 2019 from 8:00 – 10:00pm")
+    })
+
+    it("date is same day in same year as now", () => {
+      const period = datesWithHours(
+        "2018-12-05T20:00:00+00:00",
+        "2018-12-05T22:00:00+00:00"
+      )
+      expect(period).toBe("Dec 5 from 8:00 – 10:00pm")
+    })
+  })
+
+  describe("datesWithDaysAndHours", () => {
+    const realNow = Date.now
+    beforeEach(() => {
+      Date.now = () => new Date("2018-01-30T03:24:00") as any
+    })
+    afterEach(() => {
+      Date.now = realNow
+    })
+
+    it("includes the start and end date with different years", () => {
+      const period = datesWithDaysAndHours(
+        "2021-12-05T20:00:00+00:00",
+        "2022-12-30T17:00:00+00:00"
+      )
+      expect(period).toBe(
+        "Sun, Dec 5, 2021 at 8:00pm – Fri, Dec 30, 2022 at 5:00pm"
```

Comment:
Missing the TZ component https://www.notion.so/artsy/Formatting-Time-f6f4aaeeba414583a207013fec5a9794#4e5dea1d5229420793919615a3e4b4be